connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -297,6 +301,55 @@ public synchronized HerderRequest restartConnector(long delayMs, final String co
         return new StandaloneHerderRequest(requestSeqNum.incrementAndGet(), future);
     }
 
+    @Override
+    public synchronized void restartConnectorAndTasks(RestartRequest request, Callback<ConnectorStateInfo> cb) {
+        // Ensure the connector exists
+        String connectorName = request.connectorName();
+        if (!configState.contains(connectorName)) {
+            cb.onCompletion(new NotFoundException("Unknown connector: " + connectorName, null), null);
+            return;
+        }
+
+        Optional<RestartPlan> maybePlan = buildRestartPlanFor(request);
+        if (!maybePlan.isPresent()) {
+            cb.onCompletion(new NotFoundException("Status for connector " + connectorName + " not found", null), null);
+            return;
+        }
+        RestartPlan plan = maybePlan.get();
+
+        // If requested, stop the connector and any tasks, marking each as restarting
+        log.info("Received {}", plan);
+        if (plan.restartConnector()) {
+            worker.stopAndAwaitConnector(connectorName);
+            recordRestarting(connectorName);
+        }
+        if (plan.restartAnyTasks()) {
+            // Stop the tasks and mark as restarting
+            worker.stopAndAwaitTasks(plan.taskIdsToRestart());
+            plan.taskIdsToRestart().forEach(this::recordRestarting);
+        }
+
+        // Now restart the connector and tasks
+        if (plan.restartConnector()) {
+            log.debug("Restarting connector '{}'", connectorName);
+            startConnector(connectorName, (error, targetState) -> {
+                if (error == null) {
+                    log.debug("Connector '{}' successfully restarted", connectorName);
+                } else {
+                    log.debug("Connector '{}' failed to restart", connectorName, error);
```

Comment:
Nit: I wonder if these log messages would be a bit easier to search and filter if they were reworded slightly:
```suggestion
                    log.debug("Connector '{}' restart successful", connectorName);
                } else {
                    log.debug("Connector '{}' restart failed", connectorName, error);
```
If you agree, then maybe also make similar changes to the corresponding log messages in the `DistributedHerder`.

Reply:
+1 I just fixed them in both classes